ext/openssl/xp_ssl.c

```diff
@@ -950,7 +951,39 @@ static int php_openssl_set_local_cert(SSL_CTX *ctx, php_stream *stream) /* {{{ *
 			}
 			GET_VER_OPT_STRING("local_pk", private_key);
 
-			if (private_key) {
+			if (private_key && (strncmp(private_key, "pkcs11:", 7) == 0)) {
+				/* local_pk is a RFC7512 URI */
+				ENGINE *engine;
+				EVP_PKEY *pkey;
+				engine = ENGINE_by_id("pkcs11");
+				if (engine == NULL) {
+					php_error_docref(NULL, E_WARNING, "Could not bind PKCS11 engine");
+					return FAILURE;
+				}
+#if 0
+				if (!ENGINE_ctrl_cmd_string(engine, "VERBOSE", NULL, 0)) {
+					ENGINE_free(engine);
+					php_error_docref(NULL, E_WARNING, "PKCS11 engine VERBOSE error");
+					return FAILURE;
+				}
+#endif
```

Comment:
We don't keep a dead code so this should be deleted.

Reply:
ok, I'll remove it